cli/src/export.rs

```diff
@@ -24,17 +47,499 @@ pub struct ExportCommand {
+            .collect();
+
+        if overrides_list.len() == OVERRIDES_LIST_MAX_COUNT {
+            overrides_list.push("- ...".into());
+        }
+
+        //TODO: what happens if there's a field named override?
+        let override_arg_label = "override";
+        let override_help = format!(
+            "Override any field of the configuration with a valid Nickel expression provided as \
+            a string. The new value will be merged with the configuration with a `force` \
+            priority.\
+            \n\nOverridable fields:\n{}\n",
+            overrides_list.join("\n")
+        );
+        let override_arg = clap::Arg::new(override_arg_label)
+            .long(override_arg_label)
+            .value_name(NICKEL_VALUE_NAME.to_owned())
+            .number_of_values(2)
+            .value_names(["field", "value"])
+            .action(ArgAction::Append)
+            .required(false)
+            .help(override_help);
+
+        cmd = cmd.arg(override_arg);
+
+        TermCommand {
+            clap_cmd: cmd,
+            args: paths,
+            overrides: ovd_interfaces,
+        }
+    }
+}
+
+impl Combine for TermInterface {
+    fn combine(first: Self, second: Self) -> Self {
+        let TermInterface { mut fields } = first;
+
+        for (id, field) in second.fields.into_iter() {
+            if let Some(prev) = fields.remove(&id) {
+                fields.insert(id, Combine::combine(prev, field));
+            } else {
+                fields.insert(id, field);
+            }
+        }
+
+        TermInterface { fields }
+    }
+}
+
+impl Combine for FieldInterface {
+    fn combine(first: Self, second: Self) -> Self {
+        FieldInterface {
+            subfields: Combine::combine(first.subfields, second.subfields),
+            field: Field {
+                metadata: Combine::combine(first.field.metadata, second.field.metadata),
+                // Value is used only to show a default value, and to determine if a field has a
+                // definition. We don't bother actually merging the content, but just keep any
+                // side that is defined.
+                value: first.field.value.or(second.field.value),
+                ..Default::default()
+            },
+        }
+    }
+}
+
+impl From<&RecordData> for TermInterface {
+    fn from(value: &RecordData) -> Self {
+        TermInterface {
+            fields: value
+                .fields
+                .iter()
+                .map(|(id, field)| (*id, field.into()))
+                .collect(),
+        }
+    }
+}
+
+impl From<&Term> for TermInterface {
+    fn from(term: &Term) -> Self {
+        term.extract_interface().unwrap_or_else(TermInterface::new)
+    }
+}
+
+trait ExtractInterface {
+    fn extract_interface(&self) -> Option<TermInterface>;
+}
+
+impl ExtractInterface for &Term {
+    fn extract_interface(&self) -> Option<TermInterface> {
+        if let Term::Record(rd) = self {
```

Comment:
Does `RecRecord` ever happen here, or have those all been turned into `Record` by this point?

Reply:
`RecRecord` shouldn't be possible here, because we do some evaluation (the "record spine form") before extracting the interface. The only unevaluated forms are expressions whose evaluation depends on a field that doesn't have a definition yet. But this is impossible for a `RecRecord`: the latter always evaluates to a `Record` in one step.